src/lib/net/encryption/src/key.rs

```diff
@@ -0,0 +1,57 @@
+use rsa::{RsaPublicKey, RsaPrivateKey, pkcs1::EncodeRsaPublicKey, Pkcs1v15Encrypt};
+use rand::rngs::OsRng;
+use crate::errors;
+
+pub struct KeyPair {
+    pub public_key: RsaPublicKey, //public key side of the keypair (for encryption)
+    pub encoded_public_key: der::Document, //the public key encoded in ASN.1 DER format for publishing to the client
+    pub private_key: RsaPrivateKey, //private key side of the keypair (for decryption)
```

Comment:
When documenting field the actual syntax should be:
```rust
/// public key side of the keypair (for encryption)
pub public_key: RsaPublicKey
```
As it ties the documentation comment to the field that is then viewable with Rust-analyzer/LSP